### PR TITLE
added addThemAll file

### PR DESCRIPTION
**Pull request by Anatorusan**

### CODE REVIEW

File: advancedFunctions/addThemAll.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const addThemAll = (...args) => {
+    return args.reduce((arg1, arg2) => arg1 + arg2);
```
Comment (sharkich) [inline advancedFunctions/addThemAll.js]: no need `return` statement.
add default value for reduce, plz

File: advancedFunctions/byProperty.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    const compareFunc = (a, b) => {
+      if (a[property] < b[property]) {
+        return -1 * modif[direction];
+      }
+      if (a[property] > b[property]) {
+        return 1 * modif[direction];
+      }
+      return 0;
+    }
+    return compareFunc;
```
Comment (sharkich) [inline advancedFunctions/byProperty.js]: I know it's working but it's hard to understand.
Let's create a simple ordering function in one direction and then you can reverse it if it's necessary.

File: advancedFunctions/decorator.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    for(let count = 3; count < n; count ++) {
+        arr.push(arr.slice(-3).reduce((a, b) => a + b));
+      }
+    arr.slice(0, n).forEach((element) => console.log(element));
+  }
+
+const slower = (func, seconds) => {
+  const wrapper = (...args) => {
+    console.log(`Chill out bro, you will get you result in ${seconds} seconds`);
+    return setTimeout(func, seconds*1000, ...args);
```
Comment (sharkich) [inline advancedFunctions/decorator.js]: this is good that you know the third param in the setTimeout but let's use the `apply` or `call` methods.

File: advancedFunctions/decorator.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const tribonacci = (signature,n) => {
+    const arr = signature;
+    for(let count = 3; count < n; count ++) {
+        arr.push(arr.slice(-3).reduce((a, b) => a + b));
```
Comment (sharkich) [inline advancedFunctions/decorator.js]: default value of the reduce

File: advancedFunctions/decorator.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+const tribonacci = (signature,n) => {
+    const arr = signature;
```
Comment (sharkich) [inline advancedFunctions/decorator.js]:
```suggestion
    const arr =[...signature];
```

File: advancedFunctions/detonatorTimer.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+//setTimout recursion
+const detonatorTimerRecr = (delay) => {
+    if (delay < 0) {
+      return
+    }
+    console.log(delay === 0 ? 'BOOM!' : delay);
+    delay -= 1;
+    setTimeout(detonatorTimerRecr, 1000, delay);
```
Comment (sharkich) [inline advancedFunctions/detonatorTimer.js]: do not use the third param in the setTimeout

File: advancedFunctions/detonatorTimer.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    delay -= 1;
+    setTimeout(counter, 1000, delay);
+    }, 1000, delay)
+  }
+
+//setInterval
+const detonatorTimerInterv = (delay) => {
+    const timerId = setInterval(() => {
+      console.log(delay === 0 ? 'BOOM!' : delay);
+      delay -= 1;
```
Comment (sharkich) [inline advancedFunctions/detonatorTimer.js]: do not change the input

File: advancedFunctions/detonatorTimer.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    if (delay < 0) {
+      return
+    }
+    console.log(delay === 0 ? 'BOOM!' : delay);
+    delay -= 1;
+    setTimeout(detonatorTimerRecr, 1000, delay);
+  }
+
+//setTimout nested
+const detonatorTimerNest = (delay) => {
```
Comment (sharkich) [inline advancedFunctions/detonatorTimer.js]: you need to choose a version of your solution

**Review by sharkich — CHANGES_REQUESTED** [review]

File: advancedFunctions/detonatorTimer.js (hunk 8 above)
Comment (Anatorusan) [reply advancedFunctions/detonatorTimer.js]: The nested solution left

File: advancedFunctions/detonatorTimer.js (hunk 7 above)
Comment (Anatorusan) [reply advancedFunctions/detonatorTimer.js]: aded a variable for this purpose

File: advancedFunctions/detonatorTimer.js (hunk 6 above)
Comment (Anatorusan) [reply advancedFunctions/detonatorTimer.js]: Done

File: advancedFunctions/decorator.js (hunk 5 above)
Comment (Anatorusan) [reply advancedFunctions/decorator.js]: Done

File: advancedFunctions/decorator.js (hunk 4 above)
Comment (Anatorusan) [reply advancedFunctions/decorator.js]: Added 0

File: advancedFunctions/decorator.js (hunk 3 above)
Comment (Anatorusan) [reply advancedFunctions/decorator.js]: Done

File: advancedFunctions/byProperty.js (hunk 2 above)
Comment (Anatorusan) [reply advancedFunctions/byProperty.js]: Done

File: advancedFunctions/addThemAll.js (hunk 1 above)
Comment (Anatorusan) [reply advancedFunctions/addThemAll.js]: Done

File: advancedFunctions/addThemAll.js [hunk 9]
```diff
@@ -0,0 +1 @@
+const addThemAll = (...args) => args.reduce((arg1, arg2) => arg1 + arg2, 0);
```
Comment (sharkich) [inline advancedFunctions/addThemAll.js:1]: cool name!)

File: advancedFunctions/byProperty.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    },
+    ];
+
+const byProperty = (property, direction) => {
+  const compareFunc = (a, b) => {
+    if (a[property] < b[property]) {
+      return direction === '>' ? -1 : 1;
+    }
+    return direction === '>' ? 1 : -1;
+  }
```
Comment (sharkich) [inline advancedFunctions/byProperty.js:37]: no returning for equal properties (zero).

File: advancedFunctions/byProperty.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    runningTimeInMinutes: 107,
+    },
+    ];
+
+const byProperty = (property, direction) => {
+  const compareFunc = (a, b) => {
+    if (a[property] < b[property]) {
+      return direction === '>' ? -1 : 1;
+    }
+    return direction === '>' ? 1 : -1;
```
Comment (sharkich) [inline advancedFunctions/byProperty.js]: copy/paste from line above

File: advancedFunctions/detonatorTimer.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    let countDown = delay;
+    setTimeout(function counter () {
```
Comment (sharkich) [inline advancedFunctions/detonatorTimer.js]: arrow function

**Review by sharkich — CHANGES_REQUESTED** [review]

File: advancedFunctions/byProperty.js (hunk 11 above)
Comment (Anatorusan) [reply advancedFunctions/byProperty.js]: Restructured - no code repetition

File: advancedFunctions/detonatorTimer.js (hunk 12 above)
Comment (Anatorusan) [reply advancedFunctions/detonatorTimer.js]: Added arrow function

File: advancedFunctions/byProperty.js (hunk 10 above)
Comment (Anatorusan) [reply advancedFunctions/byProperty.js:37]: Added case for 0

File: advancedFunctions/detonatorTimer.js [hunk 13]
```diff
@@ -0,0 +1,25 @@
+//setTimout nested
+const detonatorTimerNest = (delay) => {
+    let countDown = delay;
+    const counter = () => {
+      if (countDown < 0) {
+        return;
+      }
+      console.log(countDown === 0 ? 'BOOM!' : countDown);
+      countDown -= 1;
+      setTimeout(counter, 1000);
```
Comment (sharkich) [inline advancedFunctions/detonatorTimer.js]: copy/paste with line 12. it's better to use recursion then.

**Review by sharkich — APPROVED** [review]

**Pull request merged**